tests/suite/snippets_filter_test.go

```diff
@@ -0,0 +1,340 @@
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%shttp.server_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+						Servers:   []string{"cafe.example.com"},
+						File:      "http.conf",
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%shttp.server.location_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+						File:      "http.conf",
+						Location:  "/coffee",
+						Servers:   []string{"cafe.example.com"},
+					},
+				}),
+				Entry("GRPCRoute", []framework.ExpectedNginxField{
+					{
+						Directive: "worker_shutdown_timeout",
+						Value:     "120s",
+						File:      fmt.Sprintf("%s%smain_%s", filePrefix, snippetsFilterPrefix, grpcRoutePrefix),
+					},
+					{
+						Directive: "types_hash_bucket_size",
+						Value:     "64",
+						File:      fmt.Sprintf("%s%shttp_%s", filePrefix, snippetsFilterPrefix, grpcRoutePrefix),
+					},
+					{
+						Directive: "server_tokens",
+						Value:     "on",
+						File:      fmt.Sprintf("%s%shttp.server_%s", filePrefix, snippetsFilterPrefix, grpcRoutePrefix),
+					},
+					{
+						Directive: "tcp_nodelay",
+						Value:     "on",
+						File:      fmt.Sprintf("%s%shttp.server.location_%s", filePrefix, snippetsFilterPrefix, grpcRoutePrefix),
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%shttp.server_%s", filePrefix, snippetsFilterPrefix, grpcRoutePrefix),
+						Servers:   []string{"*.example.com"},
+						File:      "http.conf",
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%shttp.server.location_%s", filePrefix, snippetsFilterPrefix, grpcRoutePrefix),
+						File:      "http.conf",
+						Location:  "/helloworld.Greeter/SayHello",
+						Servers:   []string{"*.example.com"},
+					},
+				}),
+			)
+		})
+	})
+
+	When("SnippetsFilter is invalid", func() {
+		Specify("if directives already present in the config are used", func() {
+			files := []string{"snippets-filter/invalid-duplicate-sf.yaml"}
+
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			nsname := types.NamespacedName{Name: "tea", Namespace: namespace}
+			Expect(waitForHTTPRouteToHaveGatewayNotProgrammedCond(nsname)).To(Succeed())
+
+			Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+		})
+
+		Specify("if directives are provided in the wrong context", func() {
+			files := []string{"snippets-filter/invalid-context-sf.yaml"}
+
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			nsname := types.NamespacedName{Name: "soda", Namespace: namespace}
+			Expect(waitForHTTPRouteToHaveGatewayNotProgrammedCond(nsname)).To(Succeed())
+
+			Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+		})
+	})
+})
+
+func waitForHTTPRouteToHaveGatewayNotProgrammedCond(httpRouteNsNames types.NamespacedName) error {
```

Comment:
This function isn't doing anything specific to what the title suggests. It just creates a context and calls another function. Doesn't really seem necessary. You could just define the context in the caller.

Reply:
Actually, maybe we just move the `waitForHTTPRouteStatus` into this function instead. I don't think we need the separation. Same for the snippets status check below.